pytools/test/test_persistent_dict.py

```diff
@@ -528,6 +528,16 @@ def update_persistent_hash(self, key_hash, key_builder):
 def test_class_hashing() -> None:
     keyb = KeyBuilder()
 
+    class WithoutUpdateMethod:
+        pass
+
+    assert keyb(WithoutUpdateMethod) == keyb(WithoutUpdateMethod)
```

Comment:
Arguably, this should not work. This is a different class every time the function gets called. It might be useful to see if the class type has the same `id()` as the one available at the advertised name, before using that name for hashing.

Reply:
TIL, thanks! What do you think of e35c58a? I'm worried that this may cause extra cache misses since some tags that we use might be local classes?